### ISSUE TITLE
Playing tarck after fefreshing playlist results in app terminating due to uncaught exception

### ISSUE THREAD

**Issue by ccp-codex** [BLeeEZ/amperfy#6]
Full details below. This happens when I attempt to play a track for the first time after refreshing a playlist. The relevant bit appears here : 

2019-06-10 13:11:49.216194+0000 Amperfy[2675:112209] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Illegal attempt to establish a relationship 'song' between objects in different contexts (source = <PlaylistElement: 0x600000d094f0>

```
2019-06-10 13:03:43.422449+0000 Amperfy[2528:105604] [BackendProxy] Ampache is active backend api
2019-06-10 13:03:43.422778+0000 Amperfy[2528:105604] [BackgroundSyncer] SyncInBackground start
2019-06-10 13:03:43.425361+0000 Amperfy[2528:105766] [Ampache] https://cloud.redshift.is/apps/music/ampache/server/xml.server.php?action=handshake&auth=f5a8972f9938f78ff4f557a862f3865ea5fc5c1ddb2a0339973c2106e55c5b48&timestamp=1560171823&version=350001&user=ncp
2019-06-10 13:03:43.428708+0000 Amperfy[2528:105777] [Ampache] https://cloud.redshift.is/apps/music/ampache/server/xml.server.php?action=handshake&auth=f5a8972f9938f78ff4f557a862f3865ea5fc5c1ddb2a0339973c2106e55c5b48&timestamp=1560171823&version=350001&user=ncp
2019-06-10 13:03:43.902051+0000 Amperfy[2528:105777] [LibrarySyncer] Lib resync: New changes on server
2019-06-10 13:03:43.906400+0000 Amperfy[2528:105777] [LibrarySyncer] Lib resync: 0 Artists parsed
2019-06-10 13:03:44.204496+0000 Amperfy[2528:105777] [Parser] Found album id 20 with unknown artist 39
2019-06-10 13:03:44.206156+0000 Amperfy[2528:105777] [Parser] Found album id 29 with unknown artist 57
2019-06-10 13:03:44.207338+0000 Amperfy[2528:105777] [Parser] Found album id 35 with unknown artist 41
2019-06-10 13:03:44.208440+0000 Amperfy[2528:105777] [Parser] Found album id 21 with unknown artist 41
2019-06-10 13:03:44.209711+0000 Amperfy[2528:105777] [Parser] Found album id 25 with unknown artist 41
2019-06-10 13:03:44.211076+0000 Amperfy[2528:105777] [Parser] Found album id 22 with unknown artist 41
2019-06-10 13:03:44.212787+0000 Amperfy[2528:105777] [Parser] Found album id 34 with unknown artist 41
2019-06-10 13:03:44.214312+0000 Amperfy[2528:105777] [Parser] Found album id 27 with unknown artist 41
2019-06-10 13:03:44.215627+0000 Amperfy[2528:105777] [Parser] Found album id 28 with unknown artist 41
2019-06-10 13:03:44.216975+0000 Amperfy[2528:105777] [Parser] Found album id 23 with unknown artist 41
2019-06-10 13:03:44.218504+0000 Amperfy[2528:105777] [Parser] Found album id 24 with unknown artist 41
2019-06-10 13:03:44.219798+0000 Amperfy[2528:105777] [Parser] Found album id 26 with unknown artist 41
2019-06-10 13:03:44.220200+0000 Amperfy[2528:105777] [LibrarySyncer] Lib resync: 12 Albums parsed
2019-06-10 13:03:44.590939+0000 Amperfy[2528:105777] [Parser] Found song id 22 with unknown artist 41
2019-06-10 13:03:44.595233+0000 Amperfy[2528:105777] [Parser] Found song id 21 with unknown artist 41
2019-06-10 13:03:44.597298+0000 Amperfy[2528:105777] [Parser] Found song id 23 with unknown artist 41
2019-06-10 13:03:44.599158+0000 Amperfy[2528:105777] [Parser] Found song id 35 with unknown artist 41
2019-06-10 13:03:44.601130+0000 Amperfy[2528:105777] [Parser] Found song id 36 with unknown artist 57
2019-06-10 13:03:44.602950+0000 Amperfy[2528:105777] [Parser] Found song id 32 with unknown artist 57
2019-06-10 13:03:44.603958+0000 Amperfy[2528:105777] [Parser] Found song id 24 with unknown artist 41
2019-06-10 13:03:44.605519+0000 Amperfy[2528:105777] [Parser] Found song id 25 with unknown artist 41
2019-06-10 13:03:44.607175+0000 Amperfy[2528:105777] [Parser] Found song id 26 with unknown artist 41
2019-06-10 13:03:44.609061+0000 Amperfy[2528:105777] [Parser] Found song id 27 with unknown artist 41
2019-06-10 13:03:44.610666+0000 Amperfy[2528:105777] [Parser] Found song id 28 with unknown artist 41
2019-06-10 13:03:44.613099+0000 Amperfy[2528:105777] [Parser] Found song id 31 with unknown artist 57
2019-06-10 13:03:44.614204+0000 Amperfy[2528:105777] [Parser] Found song id 30 with unknown artist 57
2019-06-10 13:03:44.615306+0000 Amperfy[2528:105777] [Parser] Found song id 29 with unknown artist 57
2019-06-10 13:03:44.616334+0000 Amperfy[2528:105777] [Parser] Found song id 33 with unknown artist 57
2019-06-10 13:03:44.617374+0000 Amperfy[2528:105777] [Parser] Found song id 34 with unknown artist 41
2019-06-10 13:03:44.618902+0000 Amperfy[2528:105777] [Parser] Found song id 20 with unknown artist 39
2019-06-10 13:03:44.619492+0000 Amperfy[2528:105777] [LibrarySyncer] Lib resync: 17 Songs parsed
2019-06-10 13:03:44.685905+0000 Amperfy[2528:105766] [LibrarySyncer] All artworks are synced
2019-06-10 13:04:16.120482+0000 Amperfy[2528:105604] Play song: Unknown artist - Edge Of The Dream
2019-06-10 13:04:40.418412+0000 Amperfy[2528:105604] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x60000291fd90 UIView:0x7fcf0793a200.width == - 16   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000291fd90 UIView:0x7fcf0793a200.width == - 16   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.
2019-06-10 13:05:28.136731+0000 Amperfy[2528:105604] [AXMediaCommon] Unable to look up screen scale
2019-06-10 13:05:28.136827+0000 Amperfy[2528:105604] [AXMediaCommon] Unexpected physical screen orientation
2019-06-10 13:05:28.160506+0000 Amperfy[2528:105604] [AXMediaCommon] Unable to look up screen scale
2019-06-10 13:05:28.163483+0000 Amperfy[2528:105604] [AXMediaCommon] Unable to look up screen scale
2019-06-10 13:05:28.163574+0000 Amperfy[2528:105604] [AXMediaCommon] Unexpected physical screen orientation
2019-06-10 13:05:49.859681+0000 Amperfy[2528:107169] [AmpacheLibSyncer] Download playlist "rosecross" from server
2019-06-10 13:05:50.207395+0000 Amperfy[2528:107169] [AmpacheLibSyncer] Sync songs of playlist "rosecross"
2019-06-10 13:05:52.264957+0000 Amperfy[2528:105604] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Illegal attempt to establish a relationship 'song' between objects in different contexts (source = <PlaylistElement: 0x6000029b68f0> (entity: PlaylistElement; id: 0x600000aaaee0 <x-coredata:///PlaylistElement/tC3B05E8F-E8B0-4FA2-85AF-6EDB2C70769B8> ; data: {
    order = 0;
    playlist = "0xc11353cbb0b2bf20 <x-coredata://9BE5A25C-7615-4BA3-A96D-733D73C4AB8D/PlaylistManaged/p2>";
    song = nil;
}) , destination = <Song: 0x6000029b5f90> (entity: Song; id: 0xc11353cbb0cabf2a <x-coredata://9BE5A25C-7615-4BA3-A96D-733D73C4AB8D/Song/p28> ; data: {
    album = "0xc11353cbb0a2bf2e <x-coredata://9BE5A25C-7615-4BA3-A96D-733D73C4AB8D/Album/p6>";
    artist = nil;
    artwork = nil;
    dataMO = nil;
    id = 36;
    playlistElements =     (
        "0xc11353cbb09ebf24 <x-coredata://9BE5A25C-7615-4BA3-A96D-733D73C4AB8D/PlaylistElement/p9>"
    );
    syncInfo = "0xc11353cbb0b2bf26 <x-coredata://9BE5A25C-7615-4BA3-A96D-733D73C4AB8D/SyncWaveMO/p2>";
    title = Decani;
    track = 6;
    url = "https://cloud.redshift.is/apps/music/ampache/server/xml.server.php?action=play&filter=36&auth=a10c5313a78248ba469522ed9aa93285";
}))'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010649d6fb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x0000000104ca7ac5 objc_exception_throw + 48
	2   CoreData                            0x0000000105e613a1 _PFManagedObject_coerceValueForKeyWithDescription + 2481
	3   CoreData                            0x0000000105e3de15 _sharedIMPL_setvfk_core + 229
	4   Amperfy                             0x00000001040d4d86 $s7Amperfy8PlaylistC6append4songyAA4SongC_tF + 582
	5   Amperfy                             0x00000001040c74a4 $s7Amperfy6PlayerC13addToPlaylist4songyAA4SongC_tF + 84
	6   Amperfy                             0x00000001040c7e39 $s7Amperfy6PlayerC4play4songyAA4SongC_tF + 105
	7   Amperfy                             0x000000010415a94b $s7Amperfy13SongTableCellC12touchesEnded_4withyShySo7UITouchCG_So7UIEventCSgtF + 507
	8   Amperfy                             0x000000010415ab5c $s7Amperfy13SongTableCellC12touchesEnded_4withyShySo7UITouchCG_So7UIEventCSgtFTo + 124
	9   UIKitCore                           0x000000010fcc0a09 forwardTouchMethod + 353
	10  UIKitCore                           0x000000010fcc0ad8 -[UIResponder touchesEnded:withEvent:] + 49
	11  UIKitCore                           0x000000010f8789c9 _UIGestureEnvironmentUpdate + 10367
	12  UIKitCore                           0x000000010f876108 -[UIGestureEnvironment _deliverEvent:toGestureRecognizers:usingBlock:] + 478
	13  UIKitCore                           0x000000010f875e96 -[UIGestureEnvironment _updateForEvent:window:] + 200
	14  UIKitCore                           0x000000010fcd15bc -[UIWindow sendEvent:] + 4057
	15  UIKitCore                           0x000000010fcafd16 -[UIApplication sendEvent:] + 356
	16  UIKit                               0x00000001289c14af -[UIApplicationAccessibility sendEvent:] + 85
	17  UIKitCore                           0x000000010fd80293 __dispatchPreprocessedEventFromEventQueue + 3232
	18  UIKitCore                           0x000000010fd82bb9 __handleEventQueueInternal + 5911
	19  CoreFoundation                      0x0000000106404be1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	20  CoreFoundation                      0x0000000106404463 __CFRunLoopDoSources0 + 243
	21  CoreFoundation                      0x00000001063feb1f __CFRunLoopRun + 1231
	22  CoreFoundation                      0x00000001063fe302 CFRunLoopRunSpecific + 626
	23  GraphicsServices                    0x000000010c1742fe GSEventRunModal + 65
	24  UIKitCore                           0x000000010fc95ba2 UIApplicationMain + 140
	25  Amperfy                             0x0000000104192f6b main + 75
	26  libdyld.dylib                       0x00000001098ab541 start + 1
	27  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb)
```

**Comment by BLeeEZ** [comment]
I can reproduce this issue on my machine too.

**Comment by BLeeEZ** [comment]
Fixed with a79635c2b7fe86481ad846211672e151784ff616
